### ISSUE TITLE
Genes not found, function fails

### ISSUE THREAD

**Issue by hilldr** [hilldr/gene_retriever#14]
If a user submits a list of genes and any one of these genes is not found in the dataset, 'gene retriever' will fail to generate output. This is inconvenient when querying large lists of genes or forwarding lists from other R functions.

Gene retriever should be able to generate output for genes that are found and return to the user a list of genes that were not found

**Comment by hilldr** [comment]
New code for subsetting based on a list of terms

``` R
# retrieve index of rows containing gene names in the K-means group 3 subset
group3fpkm <- which(completefpkm$gene_short_name %in% rownames(gp3)) 
# Create data frame of fpkm values for group 3 genes
gp3.fpkm.table <- completefpkm[group3fpkm,]
```

**Comment by hilldr** [comment]
This was not working using the "output_format" branch on 06/16/2015

Here is the working R function call (with the correct gene names)

``` R
gene_retriever_png(gene_names=c("ADORA2B","BNIP3","BNIP3L","PFKFB3","LDHA","SLC2A1","CKM"),nrow=2,dir="./norm_out",csv.out="./RESULTS/gr_output.csv",png="./RESULTS/grout.png")

```

Change any gene name, and the following error appears:

``` R
+ + + Error in `$<-.data.frame`(`*tmp*`, "weight", value = 1) : 
  replacement has 1 row, data has 0
Error in seq.default(from = best$lmin, to = best$lmax, by = best$lstep) (from gr_ouputformat.r#175) : 
  'from' must be of length 1
In addition: Warning message:
In loop_apply(n, do.ply) :
  Removed 9 rows containing non-finite values (stat_boxplot).
```

**Comment by sthodla** [comment]
I ran the working function call, but it is telling me that it could not find any of the genes and I can't find the problem. Are you using a different gene file?

**Comment by hilldr** [comment]
Oh, yes. This is using an unpublished "./norm_out" format data directory. Does it work if you use other gene names that you know are in the data directory? Find a set that works and then "misspell" one or two and see if you get an error or if it works

**Comment by sthodla** [comment]
It works for me when I misspell two. Gives me the names of the two genes it could not find and exports the output for the rest.

**Comment by hilldr** [comment]
I was using the 'gene-retriever_png' option if that makes a difference

On Tue, 16 Jun 2015 12:22:46 -0700
sthodla notifications@github.com wrote:

> It works for me when I misspell two. Gives me the names of the two genes it could not find and exports the output for the rest.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/hilldr/gene_retriever/issues/14#issuecomment-112535551

## 

David R. Hill, PhD

Postdoctoral Fellow
Department of Internal Medicine
Division of Gastroenterology
University of Michigan Medical Center
1150 W. Medical Center Dr.
Ann Arbor MI 48109

Phone: 440.832.0000
Email: hilldr@med.umich.edu

---

Electronic Mail is not secure, may not be read every day, and should not be used for urgent or sensitive issues

**Comment by sthodla** [comment]
Oh, I think I forgot to update the png function, I'll do that now and push it.

**Comment by sthodla** [comment]
It should work now

**Comment by hilldr** [comment]
Fixed in new SeqRetriever v0.3-alpha
